src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/SqlAction.cs

```diff
@@ -480,9 +376,28 @@ NpgsqlTransaction transaction
         int[] parentDocumentPartitionKeys = new int[bulkReferences.ReferentialIds.Length];
         Array.Fill(parentDocumentPartitionKeys, bulkReferences.ParentDocumentPartitionKey);
 
+        // Use unnest() to bulk insert references, left join to find referenced documents ids
+        // or null if this is an invalid reference (and FK constraint is disabled)
         await using var command = new NpgsqlCommand(
-            @"INSERT INTO dms.Reference (ParentDocumentId, ParentDocumentPartitionKey, ReferentialId, ReferentialPartitionKey)
-                    SELECT * FROM unnest($1, $2, $3, $4)",
+            @"INSERT INTO dms.Reference (
+                  ParentDocumentId,
+                  ParentDocumentPartitionKey,
+                  ReferentialId,
+                  ReferentialPartitionKey,
+                  ReferencedDocumentId,
+                  ReferencedDocumentPartitionKey
+              )
+              SELECT
+                  ids.documentId,
+                  ids.documentPartitionKey,
+                  ids.referentialId,
+                  ids.referentialPartitionKey,
+                  a.documentId,
+                  a.documentPartitionKey
+              FROM unnest($1, $2, $3, $4) AS
+                  ids(documentId, documentPartitionKey, referentialId, referentialPartitionKey)
+              LEFT JOIN dms.Alias a ON
+                  ids.referentialId = a.referentialId and ids.referentialPartitionKey = a.referentialPartitionKey",
```

Comment:
Inserts into references table now requires a join to get documentId of reference.